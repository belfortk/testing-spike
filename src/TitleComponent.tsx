import React from "react";

interface IProps{
  text: string;
  otherText: string;
}
const TitleComponent = ({text, otherText}: IProps) =>{
  return <h1 className="title">{text} {otherText}</h1>
}

export default TitleComponent;