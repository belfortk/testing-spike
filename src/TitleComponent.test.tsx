import React from 'react';
import {shallow} from 'enzyme';
import TitleComponent from './TitleComponent';

describe("Title Component", () => {
  it("it should render h1", () => {
    const wrapper = shallow(<TitleComponent text={"Hello"} otherText={"World"}/>);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('h1')).toHaveLength(1);
    expect(wrapper.text()).toMatch("Hello World")
    wrapper.setProps({text:"Goodbye"})
    expect(wrapper.text()).toMatch("Goodbye World")
    wrapper.unmount();
  });
});
