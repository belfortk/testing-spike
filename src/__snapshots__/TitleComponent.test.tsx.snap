// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component it should render h1 1`] = `ShallowWrapper {}`;
